From 35c1af372229d019ae4b4a14f328f5eeee5bad42 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 21 Sep 2021 12:30:20 +0800
Subject: [PATCH 04/30] Revert "drm/msm/mdp4: refactor HW revision detection
 into read_mdp_hw_revision"

This reverts commit 90363618b5529ae6c70dffd88bfcacaef5676cc7.
---
 drivers/gpu/drm/msm/disp/mdp4/mdp4_kms.c | 27 ++++++++----------------
 1 file changed, 9 insertions(+), 18 deletions(-)

diff --git a/drivers/gpu/drm/msm/disp/mdp4/mdp4_kms.c b/drivers/gpu/drm/msm/disp/mdp4/mdp4_kms.c
index b73af9ddcf72..2f75e3905202 100644
--- a/drivers/gpu/drm/msm/disp/mdp4/mdp4_kms.c
+++ b/drivers/gpu/drm/msm/disp/mdp4/mdp4_kms.c
@@ -19,13 +19,20 @@ static int mdp4_hw_init(struct msm_kms *kms)
 {
 	struct mdp4_kms *mdp4_kms = to_mdp4_kms(to_mdp_kms(kms));
 	struct drm_device *dev = mdp4_kms->dev;
-	u32 major, minor, dmap_cfg, vg_cfg;
+	uint32_t version, major, minor, dmap_cfg, vg_cfg;
 	unsigned long clk;
 	int ret = 0;
 
 	pm_runtime_get_sync(dev->dev);
 
-	read_mdp_hw_revision(mdp4_kms, &major, &minor);
+	mdp4_enable(mdp4_kms);
+	version = mdp4_read(mdp4_kms, REG_MDP4_VERSION);
+	mdp4_disable(mdp4_kms);
+
+	major = FIELD(version, MDP4_VERSION_MAJOR);
+	minor = FIELD(version, MDP4_VERSION_MINOR);
+
+	DBG("found MDP4 version v%d.%d", major, minor);
 
 	if (major != 4) {
 		DRM_DEV_ERROR(dev->dev, "unexpected MDP version: v%d.%d\n",
@@ -402,22 +409,6 @@ static int modeset_init(struct mdp4_kms *mdp4_kms)
 	return ret;
 }
 
-static void read_mdp_hw_revision(struct mdp4_kms *mdp4_kms,
-				 u32 *major, u32 *minor)
-{
-	struct drm_device *dev = mdp4_kms->dev;
-	u32 version;
-
-	mdp4_enable(mdp4_kms);
-	version = mdp4_read(mdp4_kms, REG_MDP4_VERSION);
-	mdp4_disable(mdp4_kms);
-
-	*major = FIELD(version, MDP4_VERSION_MAJOR);
-	*minor = FIELD(version, MDP4_VERSION_MINOR);
-
-	DRM_DEV_INFO(dev->dev, "MDP4 version v%d.%d", *major, *minor);
-}
-
 struct msm_kms *mdp4_kms_init(struct drm_device *dev)
 {
 	struct platform_device *pdev = to_platform_device(dev->dev);
-- 
2.27.0

